velox/dwio/common/IntDecoder.h

```diff
@@ -437,24 +436,28 @@ inline T IntDecoder<isSigned>::readInt() {
   if (useVInts_) {
     return readVInt<T>();
   }
+
   if (bigEndian_) {
     return readLittleEndianFromBigEndian<T>();
-  } else {
-    if constexpr (std::is_same_v<T, int128_t>) {
-      if (numBytes_ == 12) {
-        VELOX_DCHECK(!useVInts_, "Int96 should not be VInt encoded.");
-        return readInt96();
-      }
-      VELOX_NYI();
+  }
+
+  if constexpr (std::is_same_v<T, int128_t>) {
+    // Int64 or Int96 in Parquet
+    if (numBytes_ == 8 || numBytes_ == 12) {
+      return readSignedIntAs128();
     }
-    return readLongLE();
+    VELOX_NYI();
   }
+
+  return readLongLE();
 }
 
 template <bool isSigned>
-inline int128_t IntDecoder<isSigned>::readInt96() {
+inline int128_t IntDecoder<isSigned>::readSignedIntAs128() {
```

Comment:
Thus function doesn't respect isSigned, negative int64 values will not be read correctly

Reply:
Just wonder, is it possible to pass a negative value for timestamp?

Reply:
Yes, int64 timestamp is signed, negative values correspond to timestamp before 1970-01-01 00:00:00. You may read the details here - https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#timestamp

This is different from int96 timestamp which cannot be negative as it's based on julian days.